File: Languages/passwordProtect.go
package main

import "fmt"
import "os"

func part1() {
    var a int = -1
    for a < 9 {
      a = a + 1
      var string_a string = fmt.Sprintf("%d", a)
      os.Mkdir(string_a, 0755)
    }
}

func part2() {
    var a int = -1
    for a < 9 {
      a = a + 1
      var b int = -1
      for b < 9 {
        b = b + 1
        var string_ab string = fmt.Sprintf("%d/%d", a, b)
        os.Mkdir(string_ab, 0755)
      } 
    }
}

func part3() {
    var a int = -1
    for a < 9 {
      a = a + 1
      var b int = -1
      for b < 9 {
        b = b + 1
        var c int = -1
        for c < 9 {
          c = c + 1
          var string_abc string = fmt.Sprintf("%d/%d/%d", a, b, c)
          os.Mkdir(string_abc, 0755)
        } 
      } 
    }
}

func part4() {
    var a int = -1
    for a < 9 {
      a = a + 1
      var b int = -1
      for b < 9 {
        b = b + 1
        var c int = -1
        for c < 9 {
          c = c + 1
          var d int = -1
          for d < 9 {
            d = d + 1
            var string_abcd string = fmt.Sprintf("%d/%d/%d/%d", a, b, c, d)
            var filename string = fmt.Sprintf("%d/%d/%d/%d/.combo", a, b, c, d)
            os.Mkdir(string_abcd, 0755)
            os.Create(filename)
        } 
        } 
      } 
    }
}

func main() {
  part1()
  fmt.Println("25% complete")
  part2()
  fmt.Println("50% complete")
  part3()
  fmt.Println("75% complete")
  part4()
  fmt.Println("100% complete")
}

File: Languages/passwordProtect.cpp
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>
#include <sys/types.h>
#include <errno.h>

int part1(){
  for (int a = 0; a <= 9; a++) {
    char string_A[10];
    sprintf(string_A, "%d", a);
    mkdir(string_A, 0755);
  }
  
  return 0;
}

int part2(){
  for (int a = 0; a <= 9; a++) {
    for (int b = 0; b <= 9; b++) {
      char string_AB[10];
      sprintf(string_AB, "%d/%d", a, b);
      mkdir(string_AB, 0755);
    }
  }
  
  return 0;
}

int part3(){
  for (int a = 0; a <= 9; a++) {
    for (int b = 0; b <= 9; b++) {
      for (int c = 0; c <= 9; c++) {
        char string_ABC[10];
        sprintf(string_ABC, "%d/%d/%d", a, b, c);
        mkdir(string_ABC, 0755);
      }
    }
  }
  
  return 0;
}

int part4(){
  for (int a = 0; a <= 9; a++) {
    for (int b = 0; b <= 9; b++) {
      for (int c = 0; c <= 9; c++) {
        for (int d = 0; d <= 9; d++) {
        char string_ABCD[10];
        char filename[25];
        sprintf(string_ABCD, "%d/%d/%d/%d", a, b, c, d);
        sprintf(filename, "%s/.combo", string_ABCD);
        mkdir(string_ABCD, 0755);
        fopen(filename, "w");
      }
      }
    }
  }
  
  return 0;

}

int main() {
  part1();
  printf("1/4 complete\n");
  part2();
  printf("2/4 complete\n");
  part3();
  printf("3/4 complete\n");
  part4();
  printf("4/4 complete\n");
  
  return 0;
}